### PR TITLE
New features and refactor

### PR DESCRIPTION
**Pull request by vseager**
* Added support for configurable product highest and lowest price
* Added support for configurable product highest and lowest weight
* Added support for Yotpo reviews
* Include offers with the correct information for all configurable product children
* Fixed pricing when product is discounted
* Add description for CMS pages
* Improved page type declaration
* Added publisher information
* Added breadcrumbs
* Some other general refactoring

### CODE REVIEW

File: Block/Jsonld.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    public function getPageType()
+    {
+        if($this->_pageType == null) {
+            $module     = $this->_request->getModuleName();
+            $controller = $this->_request->getControllerName();
+
+            switch ($module) {
+                case 'catalog':
+                    if($controller == 'product') {
```
Comment (davidwindell) [inline Block/Jsonld.php:101]: CS

File: Block/Product.php [hunk 2]
```diff
@@ -136,14 +135,31 @@ public function getStore()
...
+        }
+
+        return false;
+    }
+
+    public function getChildren()
     {
-        return $this->_logo->getLogoSrc();
+        $configurableCode = \Magento\ConfigurableProduct\Model\Product\Type\Configurable::TYPE_CODE;
+        if(!$this->getConfig('structureddata/product/include_children') || $this->getProduct()->getTypeId() !== $configurableCode) {
```
Comment (davidwindell) [inline Block/Product.php:153]: CS

File: Block/Product.php [hunk 3]
```diff
@@ -19,6 +18,8 @@
 use Magento\Framework\Pricing\PriceCurrencyInterface;
 use Magento\Store\Model\ScopeInterface;
 use Magento\Catalog\Model\ProductFactory;
+use Magento\Framework\Module\Manager as ModuleManager;
+use Yotpo\Yotpo\Helper\RichSnippets as YotpoRichSnippets;
```
Comment (davidwindell) [inline Block/Product.php:22]: Check that Yotpo is still part of core in 2.3.5

Comment (vseager) [reply Block/Product.php:22]: https://docs.magento.com/m2/ce/user_guide/marketing/yotpo-reviews-intro.html
https://support.yotpo.com/en/article/magento-2-installing-yotpo

File: Block/Product.php (hunk 2 above)
Comment (vseager) [reply Block/Product.php]: Corrected

File: Block/Jsonld.php (hunk 1 above)
Comment (vseager) [reply Block/Jsonld.php:101]: Corrected

**Pull request merged**